Feature/tuneable shift sensor

EigenMania opened the pull request:
Shift pause duration (ms) and shift power threshold (W) can be tuned by the user through the GUI (or disabled entirely). Shift power threshold is estimated with the nominal battery voltage (also now provided through the GUI) and the maximum configured current. The threshold is computed once at init to save on computational resources while running (thus does not take into account changing battery voltages).

danielnilsson9 commented:
Thank you for working on this.

However I cannot accept the current implementation. I don't want to add unnecessary config parameters as the configuration can easily grow large and hard to understand for users. The nominal battery voltage is one of those, and it looks like you are only using it to calculate a current percentage to apply during shift sensor interrupt.

The only reason you needed the nominal battery voltage is because you chose to configure shift sensor interrupt power in Watts which is not used anywhere else.

I would prefer:
- Remove the nominal battery voltage config parameter
- Let the user configure shift sensor interrupt power in percent
- Keep the use_shift_sensor feature toggle you have added
- Keep the shift sensor interrupt duration parameter you have added

EigenMania commented:
Thank you for your feedback. 

I have implemented all of your suggestions, and squashed the commit history.

Please don't hesitate to inform me if there is anything else you would like me to address.

danielnilsson9 commented:
Looks good! I will merge this. In your opinion, the default values you have set of 600ms and and 10% current works fine? Is the similar to the behaviour of the original firmware?

EigenMania commented:
I found that it behaved quite similar to the original firmware (with the default values proposed). On a 52V setup at 10% of a max 33A draw, it would be ~170W cutoff limit. In my opinion, this should be a safe limit for derailleurs as this is a somewhat "human" level of power output. If there's consensus to change to a different default, that would be fine. But I think ultimately it'll be adjusted to personal preference and these are reasonable starting values.
